lisp/ess-inf.el

```diff
@@ -719,6 +719,19 @@ LANGUAGE is ignored."
         (error "No ESS processes running; not yet implemented to start (%s,%s)"
                language dialect)))))
 
+(defmacro with-ess--no-pop-to-buffer (&rest body)
```

Comment:
I don't think we need a whole new macro for a piece of code we only use once. If/when we need to do this more we can implement a macro then. 

I also think we should properly namespace functions/macros/variables with `ess-`.

Reply:
I don't think it's a big deal to have this internal macro, it makes the code simpler to read and understand.

We already discussed the namespacing of `with-` macros, if we are going to change the naming convention it should be in a different PR after a new discussion.

Reply:
I never said it was a big deal :-) I don't see how having a whole new macro is simpler to read or understand than a single let binding though. The name also isn't 100% accurate as we're overriding the display-buffer framework which is slightly different than pop-to-buffer. It should be something like `ess--display-buffer-no-window`.

I also don't recall settling the `with-` conversation one way or the other. The elisp manual is pretty clear that packages should namespace all functions/variables/macros with their name, especially noninteractive ones; see `(info "(elisp) Coding Conventions")`.

Reply:
> than a single let binding though.

Fully agreed because the let binding is informative through the variable name (provided it's informative). Please note that I created the macro when a single let binding was no longer sufficient. There's a with-current-buffer in there now. Surely you'd wonder what it's doing there so we'd need a comment and then the code gets stuffy and the main intent (starting a new process) no longer stands out because it's mixed with other concerns.

> The name also isn't 100% accurate as we're overriding the display-buffer framework which is slightly different than pop-to-buffer.

pop-to-buffer is in the display-buffer framework IIRC? It just has some other logic for frame, window, and buffer selection that the macro is trying to cancel out. Or is there a purely display-buffer alternative? Is that documented somewhere?

Regarding the name I think it is good because it corresponds to the pop-to-buffer calls we currently have. It is an internal name so can easily change with the implementation and does not need to be abstract, the concrete name reflecting the current implementation is actually better for now. Most of all, since it's internal we shouldn't spend too much time discussing it.

>  also don't recall settling the with- conversation one way or the other. The elisp manual is pretty clear that packages should namespace all functions/variables/macros with their name

It was in the discussion about inferior-ess- IIRC. I like `with-prefix-` because it helps reading the code, `with-` macros set up special contexts and so should stand out. I don't mind going against the elisp manual because that's how you make conventions evolve. I could be convinced to go with `prefix-with-` after surveying what other projects do (I'd rather spend my ESS time worrying about other things though).

Reply:
Quick overview of styles used in the wild. `prefix-with-` dominates.

<details>
From MELPA:

clojure-mode: `with-clojure-buffer`
use-package: `use-package--with-elapsed-timer`
evil: `evil-without-restriction` etc
company-mode: `company-capf-with-buffer`
diff-hl: `diff-hl-with-diff-switches`
elpy: `with-elpy-rpc-virtualenv-activated`
flycheck: `flycheck-error-with-buffer`
magit: `magit-with-top-level`
markdown-mode: `markdown--with-gensyms`
projectile: `projectile-with-default-dir`
smartparents: `sp-with-modes`

From Emacs:

gnus: `gnus-icalendar-with-decoded-handle`, `gnus-async-with-semaphore`
subr: `subr--with-wrapper-hook-no-warnings`
tramp: `tramp-with-demoted-errors`, `with-tramp-progress-reporter`
tramp-gvfs: `with-tramp-dbus-call-method`
rcirc: `with-rcirc-process-buffer`
soap: `soap-with-local-xmlns`
eshell: `eshell-with-file-modes`
nxml: `nxml-with-degradation-on-error`
reftex: `reftex-with-special-syntax`
table: `table-with-cache-buffer`
org: `org-with-gensyms`
server: `server-with-environment`
calc: `calc-with-default-simplification`
mode-local: `with-mode-local`
ede: `ede-with-projectfile`
semantic: `wisent-with-context`
ediff: `ediff-with-current-buffer`
vc: `with-vc-properties`
newcomment: `comment-with-narrowing`
json: `json--with-indentation`
isearch: `with-isearch-suspended`
erc: `erc-with-buffer`
js: `with-js`
xref: `xref--with-dedicated-window`
cc: `c-with-syntax-table`
python: `python-shell-with-environment`
package: `package--with-work-buffer`
backtrace: `backtrace--with-output-variables`
generator: `cps--with-value-wrapper`
</details>

Reply:
In exported symbols we have one `with-prefix-` and two `prefix-with-`. The majority of Emacs projects seems to use `prefix-with-`. So let's go with that. I have renamed the symbol.